src/components/Search/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ChangeEvent } from 'react';
+import { getCoinsList } from '../../api/api';
+import { Coin } from '../../types/types';
+import SearchIcon from '../icons/SearchIcon';
+import styles from './Search.module.scss';
+import Loader from '../Loader';
+
+const { search, search__input, search__submit } = styles;
+
+type SearchState = {
+  searchValue: string;
+  isLoading: boolean;
+};
+
+type SearchProps = {
+  updatedCoinsList: (loadedCoins: Coin[]) => void;
+};
+
+class Search extends React.Component<SearchProps, SearchState> {
+  constructor(props: SearchProps) {
+    super(props);
+    this.state = {
+      searchValue: this.getSearchValue(),
+      isLoading: false,
+    };
+  }
+
+  getSearchValue() {
+    return localStorage.getItem('searchValue') || '';
+  }
+
+  componentDidMount() {
+    this.state.searchValue
+      ? this.loadCoinsList()
+      : this.loadCoinsListFirstTime();
+  }
+
+  async loadCoinsListFirstTime() {
+    this.setState({ isLoading: true });
+    const loadedCoinsList = await getCoinsList();
+    this.setState({ isLoading: false });
+
+    if (loadedCoinsList) {
+      this.props.updatedCoinsList(loadedCoinsList.data.coins);
+    }
+  }
+
+  async loadCoinsList() {
+    this.setState({ isLoading: true });
+    const loadedCoinsList = await getCoinsList(this.state.searchValue);
+    this.setState({ isLoading: false });
+
+    if (loadedCoinsList) {
+      this.props.updatedCoinsList(loadedCoinsList.data.coins);
+    }
+
+    localStorage.setItem('searchValue', this.state.searchValue);
+  }
```

Comment:
These 2 methods have a lot in common, do you think it will make sense to create just one method with some conditionals to differentiate the behaviour based on whether the search value is empty?